### PR TITLE
prometheus-cpp: 1.1.0 -> 1.3.0, reformat, finalAttrs, fix pkgsStatic

### PR DESCRIPTION
**Pull request by tobim**
supersedes https://github.com/NixOS/nixpkgs/pull/287851.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tobim** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 packages failed to build:</summary>
  <ul>
    <li>proxysql</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>hydra</li>
    <li>hydra.doc</li>
    <li>minetest</li>
    <li>minetestserver</li>
    <li>opentelemetry-cpp</li>
    <li>opentelemetry-cpp.dev</li>
    <li>prometheus-cpp</li>
    <li>prometheus-cpp.dev</li>
  </ul>
</details>

File: pkgs/by-name/pr/prometheus-cpp/package.nix [hunk 1]
```diff
@@ -1,46 +1,54 @@
...
+  version = "1.3.0";
 
   src = fetchFromGitHub {
     owner = "jupp0r";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-qx6oBxd0YrUyFq+7ArnKBqOwrl5X8RS9nErhRDUJ7+8=";
+    repo = "prometheus-cpp";
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-XQ8N+affKVqn/hrMHWg0eN+0Op6m9ZdVNNAW0GpDAng=";
```
Comment (getchoo) [inline pkgs/by-name/pr/prometheus-cpp/package.nix:22]:
```suggestion
    tag = "v${finalAttrs.version}";
    hash = "sha256-XQ8N+affKVqn/hrMHWg0eN+0Op6m9ZdVNNAW0GpDAng=";
```
Would be nice to use these instead

**Comment by getchoo** [comment]
@ofborg build prometheus-cpp prometheus-cpp.passthru.tests

File: pkgs/by-name/pr/prometheus-cpp/package.nix [hunk 2]
```diff
@@ -41,11 +46,6 @@ stdenv.mkDerivation (finalAttrs: {
     "dev"
   ];
 
-  postInstall = ''
-    mkdir -p $dev/lib/pkgconfig
-    substituteAll ${./prometheus-cpp.pc.in} $dev/lib/pkgconfig/prometheus-cpp.pc
```
Comment (getchoo) [inline pkgs/by-name/pr/prometheus-cpp/package.nix:37]: I'm assuming this file can also be deleted now?

Comment (tobim) [reply pkgs/by-name/pr/prometheus-cpp/package.nix:37]: That slipped my mind. Thanks for the catch!

**Review by getchoo — APPROVED** [review]
Feel free to ping me in a couple days if there are no other reviews

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363342`

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>proxysql</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>hydra</li>
    <li>hydra.doc</li>
    <li>minetest</li>
    <li>minetestserver</li>
    <li>opentelemetry-cpp</li>
    <li>opentelemetry-cpp.dev</li>
    <li>prometheus-cpp</li>
    <li>prometheus-cpp.dev</li>
  </ul>
</details>

**Review by fpletz — APPROVED** [review]

**Pull request merged**